src/MgvOfferTaking.sol

```diff
@@ -306,146 +306,97 @@ abstract contract MgvOfferTaking is MgvHasOffers {
+        /* Post-execution, `sor.wants`/`sor.gives` reflect how much was sent/taken by the offer. */
+        (uint gasused, bytes32 makerData, bytes32 mgvData) = execute(pair, mor, sor);
 
-          if (mgvData == "mgv/tradeSuccess") {
-            successCount += 1;
-          }
+        require(mgvData != "mgv/tradeSuccess", "mgv/clean/offerDidNotFail");
+        require(mgvData != "mgv/notExecuted", "mgv/clean/tickTooLow");
 
-          /* In the market order, we were able to avoid stitching back offers after every `execute` since we knew a continuous segment starting at best would be consumed. Here, we cannot do this optimisation since offers in the `targets` array may be anywhere in the book. So we stitch together offers immediately after each `execute`. */
-          if (mgvData != "mgv/notExecuted") {
-            // updates best&tick
-            sor.local = dislodgeOffer(pair, sor.offer, sor.local, true);
-          }
+        /* In the market order, we were able to avoid stitching back offers after every `execute` since we knew a continuous segment starting at best would be consumed. Here, we cannot do this optimisation since offers in the `targets` array may be anywhere in the book. So we stitch together offers immediately after each `execute`. */
+        sor.local = dislodgeOffer(pair, sor.offer, sor.local, true);
 
-          /* <a id="internalSnipes/liftReentrancy"></a> Now that the current snipe is over, we can lift the lock on the book. In the same operation we
+        /* <a id="internalSnipes/liftReentrancy"></a> Now that the current snipe is over, we can lift the lock on the book. In the same operation we
         * lift the reentrancy lock, and
         * update the storage
 
         so we are free from out of order storage writes.
         */
-          sor.local = sor.local.lock(false);
-          pair.local = sor.local;
-
-          /* `payTakerMinusFees` keeps the fee in Mangrove, proportional to the amount purchased, and gives the rest to the taker */
-          payTakerMinusFees(mor, sor);
+        sor.local = sor.local.lock(false);
+        pair.local = sor.local;
 
-          /* In an inverted Mangrove, amounts have been lent by each offer's maker to the taker. We now call the taker. This is a noop in a normal Mangrove. */
-          executeEnd(mor, sor);
+        /* No fees are paid since offer execution failed. */
 
-          /* After an offer execution, we may run callbacks and increase the total penalty. As that part is common to market orders and snipes, it lives in its own `postExecute` function. */
-          if (mgvData != "mgv/notExecuted") {
-            postExecute(mor, sor, gasused, makerData, mgvData);
-          }
+        /* In an inverted Mangrove, amounts have been lent by each offer's maker to the taker. We now call the taker. This is a noop in a normal Mangrove. */
+        // FIXME: Is this needed when offer execution failed? I don't think so, but should be covered by tests
```

Comment:
Do we leave this FIXME behind?

Reply:
To answer the FIXME: `executeEnd` is a noop on Mangrove anyway but is useful on InvertedMangrove (it calls the taker contract).

Reply:
So `executeEnd` should still be invoked when an offer fails, right?